src/Microsoft.AspNetCore.Identity/SignInManager.cs

```diff
@@ -241,19 +241,64 @@ public virtual Task SignInAsync(TUser user, bool isPersistent, string authentica
                 throw new ArgumentNullException(nameof(user));
             }
 
+            var attempt = await CheckPasswordSignInAsync(user, password, lockoutOnFailure);
+            return attempt.Succeeded 
+                ? await SignInOrTwoFactorAsync(user, isPersistent)
+                : attempt;
+        }
+
+        /// <summary>
+        /// Attempts to sign in the specified <paramref name="userName"/> and <paramref name="password"/> combination
+        /// as an asynchronous operation.
+        /// </summary>
+        /// <param name="userName">The user name to sign in.</param>
+        /// <param name="password">The password to attempt to sign in with.</param>
+        /// <param name="isPersistent">Flag indicating whether the sign-in cookie should persist after the browser is closed.</param>
+        /// <param name="lockoutOnFailure">Flag indicating if the user account should be locked if the sign in fails.</param>
+        /// <returns>The task object representing the asynchronous operation containing the <see name="SignInResult"/>
+        /// for the sign-in attempt.</returns>
+        public virtual async Task<SignInResult> PasswordSignInAsync(string userName, string password,
+            bool isPersistent, bool lockoutOnFailure)
+        {
+            var user = await UserManager.FindByNameAsync(userName);
+            if (user == null)
+            {
+                return SignInResult.Failed;
+            }
+
+            return await PasswordSignInAsync(user, password, isPersistent, lockoutOnFailure);
+        }
+
+        /// <summary>
+        /// Attempts a password sign in for a user.
+        /// </summary>
+        /// <param name="user">The user to sign in.</param>
+        /// <param name="password">The password to attempt to sign in with.</param>
+        /// <param name="lockoutOnFailure">Flag indicating if the user account should be locked if the sign in fails.</param>
+        /// <returns>The task object representing the asynchronous operation containing the <see name="SignInResult"/>
+        /// for the sign-in attempt.</returns>
+        /// <returns></returns>
+        public virtual async Task<SignInResult> CheckPasswordSignInAsync(TUser user, string password, bool lockoutOnFailure)
+        {
```

Comment:
maybe i'm missing something, but i don't see where lockoutOnFailure is used... perhaps it's the diff view here in github :/